### PR TITLE
✨ Implement interruptAfterTimeLimit

### PR DESCRIPTION
**Pull request by dubzzz**
## Why is this PR for?

Related to #423

This feature will make fast-check users able to interrupt the execution of a test while it stills have job to do. Assert will report the status as if the run was supposed to stop there. If not test failed the status will be success (even if the assert has not reached the expected number of runs).

**This is a draft for a potential futur feature, not sure if the feature will reach the master**

## In a nutshell

✔️ New feature
❌ Fix an issue
❌ Documentation improvement
❌ Other: *please explain*

(✔️: yes, ❌: no)

## Potential impacts

None.

### CODE REVIEW

**Comment by dubzzz** [comment]
**What behaviour do we want in the following scenario? Are there some cases where we would like to consider it as a failure (`skipAllAfterTimeLimit` already partially covers part of the problem).**

The user asked for 100 runs by setting the value of `numRuns` but only 20 got executed (with no failure detected) before the interrupt signal.
With the current implementation, the run will be marked as successful.

File: src/fast-check-default.ts [hunk 1]
```diff
@@ -97,6 +98,7 @@ export {
   assert,
   // pre conditions
   pre,
+  interrupt,
```
Comment (dubzzz) [inline src/fast-check-default.ts]: Not sure of the need for such feature. Not sure for the naming.

**Comment by ZeRego** [comment]
When using `markInterruptAsFailure: true` can the error message contain all seeds and paths used ? Providing just the seed and path from the last run won't be enough because the problem might be in one of the previous runs.
Otherwise the developers won't have a way to reproduce the issue.

**Comment by dubzzz** [comment]
What's the best option? `markInterruptAsFailure` or `markInterruptAsSuccess`? 🤔

I think once we answer this question we can merge the PR and move on to add seed, path and others in the report 😁

File: src/fast-check-default.ts (hunk 1 above)
Comment (dubzzz) [reply src/fast-check-default.ts]: Reverting it as it might be added later in another PR

**Comment by dubzzz** [comment]
@ZeRego The flag is now released in version 1.19.0 of fast-check ;)

**Comment by 99Edorde** [comment]
> ## Why is this PR for?
> 
> 
> 
> Related to #423
> 
> 
> 
> This feature will make fast-check users able to interrupt the execution of a test while it stills have job to do. Assert will report the status as if the run was supposed to stop there. If not test failed the status will be success (even if the assert has not reached the expected number of runs).
> 
> 
> 
> **This is a draft for a potential futur feature, not sure if the feature will reach the master**
> 
> 
> 
> ## In a nutshell
> 
> 
> 
> ✔️ New feature
> 
> ❌ Fix an issue
> 
> ❌ Documentation improvement
> 
> ❌ Other: *please explain*
> 
> 
> 
> (✔️: yes, ❌: no)
> 
> 
> 
> ## Potential impacts
> 
> 
> 
> None.
> 
>
